libtransmission/announcer.cc

```diff
@@ -894,6 +894,10 @@ void on_announce_error(tr_tier* tier, char const* err, tr_announce_event e)
                 .append(fmt::format(" ({})", announce_url)));
         tier_announce_event_push(tier, e, tr_time() + interval);
     }
+    else
+    {
+        tr_logAddDebugTier(tier, fmt::format("No next tracker on announce error: {}", err));
+    }
```

Comment:
This `else` case may never happen normally:
`tier->useNextTracker();` performs a rotation of the current tracker, and we're already in `on_announce_error`, so we're already on one of the trackers, so it's not empty.

So instead of a log, we should have a `TR_ASSERT(false)` to guaranty that application quits and reports the bug.

Reply:
I added the log _just in case_, instead of doing nothing... TR_ASSERT gets stripped from release builds anyway. I may be a bit tired at the moment, but I'm not sure when the compiler doesn't report a warning in this function though. 

However, there were already a couple of checks that `tier->currentTracker()` doesn't return null: if you always have a tracker when you reach that function, aren't those useless as well? Unless I'm mistaken, once a tier gets created, trackers are immutable.

Reply:
I'll leave @ckerr to answer that.